src/transformers/pipelines/token_classification.py

```diff
@@ -270,7 +270,13 @@ def gather_pre_entities(
             if offset_mapping is not None:
                 start_ind, end_ind = offset_mapping[idx]
                 word_ref = sentence[start_ind:end_ind]
-                is_subword = len(word_ref) != len(word)
+                if getattr(self.tokenizer._tokenizer.model, "continuing_subword_prefix", None):
+                    # This is a BPE, word aware tokenizer, there is a correct way
+                    # to fuse tokens
+                    is_subword = len(word) != len(word_ref)
+                else:
+                    # This is a fallback heuristic. This will fail most likely on any kind of text + punctuation mixtures that will be considered "words". Non word aware models cannot do better than this unfortunately.
+                    is_subword = sentence[start_ind - 1 : start_ind] != " " if start_ind > 0 else False
```

Comment:
Here I would put a warning mentioning exactly what you have put above. Maybe only output once (like it's done in the tokenizer files)